### PR TITLE
fix: Make server-side filtering case-insensitive

### PR DESCRIPTION
**Pull request by tulioag**
Fixes: vaadin/flow-components#915

### CODE REVIEW

File: test/lazy-loading.html [hunk 1]
```diff
@@ -925,6 +925,38 @@
...
+              expect(comboBox.autoOpenDisabled).to.be.true;
+
+              comboBox._inputElementValue = 'iTem 12';
+              comboBox.filter = 'item 12';
+              expect(comboBox.opened).to.be.false;
+              expect(comboBox.hasAttribute('focused')).to.be.false;
+              expect(comboBox.value).to.equal('item 12');
+            });
+
+            it('should set value with multiple matches that differ only in case', () => {
```
Comment (sissbruecker) [inline test/lazy-loading.html:950]: If I understand it correctly, this test verifies that the first item is picked if there are multiple matches. I think the test name should reflect that.

File: test/lazy-loading.html [hunk 2]
```diff
@@ -925,6 +925,38 @@
               expect(comboBox.value).to.equal('item 12');
             });
 
+            it('should set value without auto-open-disabled even if case does not match', () => {
+              comboBox.autoOpenDisabled = false;
+              expect(comboBox.autoOpenDisabled).to.be.false;
+
+              comboBox._inputElementValue = 'ItEm 12';
+              comboBox.filter = 'item 12';
```
Comment (sissbruecker) [inline test/lazy-loading.html:933]: To help me understand this: `filter` is the value that the last data provider query was made for, and `_inputElementValue` is the current state of the input?

**Review by sissbruecker — CHANGES_REQUESTED** [review]

File: test/lazy-loading.html (hunk 2 above)
Comment (tulioag) [reply test/lazy-loading.html]: Thanks for pointing it out. This was wrong. They both should have the same value, as `filter` gets the value of `_inputElementValue` on the input event.

File: test/lazy-loading.html (hunk 1 above)
Comment (sissbruecker) [reply test/lazy-loading.html:954]: @tulioag What about this one?

Comment (tulioag) [reply test/lazy-loading.html:954]: Done

**Review by sissbruecker — APPROVED** [review]

**Pull request merged**